app/services/default_proxy_proxy_selector.rb

```diff
@@ -10,7 +10,7 @@ def initialize
   def available_proxies(host, provider)
     # TODO: Once we have a internal proxy marker/feature on the proxy, we can
     # swap the implementation
-    internal_proxy = ::Katello.default_capsule
+    internal_proxy = ::SmartProxy.default_capsule
```

Comment:
It's still comming from a Katello, so this call still implies dependency on `Katello` just less explicitly.
Should we instead implement notion of `internal_proxy` in the Foreman itself?